src/main/java/com/odysseusinc/arachne/datanode/service/messaging/CohortCharacterizationRequestHandler.java

```diff
@@ -1,166 +1,105 @@
-        }).orElse(null);
-    }
-
-    private MockMultipartFile generateFile(String resourcePath, String outputName) throws IOException {
-
-        ClassPathResource resource = new ClassPathResource(resourcePath);
-        final String resourceContent = IOUtils.toString(resource.getInputStream(), Charset.defaultCharset());
-        return new MockMultipartFile(outputName, ignorePreprocessingMark(resourceContent).getBytes());
-    }
-
-    private MultipartFile getRunner(String initialFileName) throws IOException {
-
-        Map<String, Object> params = new HashMap<>();
-        params.put("initialFileName", initialFileName);
-        String result = runnerTemplate.apply(params);
-        return new MockMultipartFile("main.r", result.getBytes());
-    }
-
-    @Override
-    public CommonAnalysisType getAnalysisType() {
-
-        return CommonAnalysisType.COHORT_CHARACTERIZATION;
-    }
-
-    @Override
-    public void sendResponse(List<MultipartFile> response, String id) {
-
-        centralClient.sendCommonEntityResponse(id, response.toArray(new MultipartFile[response.size()]));
-    }
+@Component
+public class CohortCharacterizationRequestHandler implements AtlasRequestHandler<CommonEntityDTO, List<MultipartFile>> {
+
+	private static final int PAGE_SIZE = 10000;
+	private static final String PACKAGE_NAME = "CohortCharacterization%d";
+	private final AtlasService atlasService;
+	private final GenericConversionService conversionService;
+	private final CommonEntityService commonEntityService;
+	private final Template runnerTemplate;
+	private final CentralSystemClient centralClient;
+
+	public CohortCharacterizationRequestHandler(AtlasService atlasService,
+																							GenericConversionService conversionService,
+																							CommonEntityService commonEntityService,
+																							@Qualifier("cohortCharacterizationTemplate") Template runnerTemplate,
+																							CentralSystemClient centralClient) {
+
+		this.atlasService = atlasService;
+		this.conversionService = conversionService;
+		this.commonEntityService = commonEntityService;
+		this.runnerTemplate = runnerTemplate;
+		this.centralClient = centralClient;
+	}
+
+	@Override
+	public List<CommonEntityDTO> getObjectsList(List<Atlas> atlasList) {
+
+		List<CohortCharacterization> ccList = atlasService.execute(atlasList, c -> c.getCohortCharacterizations(PAGE_SIZE).getContent());
+		return ccList.stream()
+						.map(cc -> conversionService.convert(cc, CommonCcShortDTO.class))
+						.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<MultipartFile> getAtlasObject(String guid) {
+
+		return commonEntityService.findByGuid(guid).map(entity -> {
+			Integer localId = entity.getLocalId();
+			List<MultipartFile> files = new ArrayList<>();
+			String packageName = String.format(PACKAGE_NAME, localId);
+			byte[] ccPackage = atlasService.execute(entity.getOrigin(), c -> c.getCohortCharacterizationPackage(entity.getLocalId(),
```

Comment:
Why would you like to import the whole R package from Atlas? I see several issues with it:
- Security - you don't know what's inside but you are going to execute it
- Network traffic - you are going to transport several MB of data each time

Instead, you could import design JSON and assemble the package in Arachne. What's wrong with the approach? We have standardized JSON definitions exactly for this.